app/models/category.py

```diff
@@ -0,0 +1,101 @@
+from sqlalchemy import String, Boolean, Integer, Text, Date, DECIMAL, ForeignKey, Enum
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from uuid import UUID
+from .base import Base
+from .evaluation import ComprehensiveEvaluation
+from .user import User
+from enum import Enum as PyEnum
+from .evaluation import RecommendedChallenge
+
+
+#  CategoryNameEnum 클래스 정의
+class CategoryNameEnum(PyEnum):
+    CODING = "코딩"
+    ENGLISH = "영어"
+    WORKOUT = "운동"
+
+
+# CategoryColorEnum 클래스 정의
+class CategoryColorEnum(PyEnum):
+    CODING = "#FDA63A"  # 호박색
+    ENGLISH = "#6C88C4"  # 인디고 블루
+    WORKOUT = "#556B2F"  # 올리브 그린
+
+
+# CategoryProgress 테이블 (한 카테고리에 여러 목표, 노트가 연결됨)
+class CategoryProgress(Base):
+    __tablename__ = "category_progresses"
+
+    user_id: Mapped[UUID] = mapped_column(
+        UUID(as_uuid=True), ForeignKey("users.id"), nullable=False
+    )
+    category_name: Mapped[CategoryNameEnum] = mapped_column(
+        Enum(CategoryNameEnum, name="category_name_enums"), nullable=False
+    )
+
+    total_goal: Mapped[int] = mapped_column(Integer, nullable=False)
+    completed_goal: Mapped[int] = mapped_column(Integer, nullable=False)
+    progress_rate: Mapped[DECIMAL] = mapped_column(DECIMAL(5, 2), default=0.00)
+
+    # CategoryProgress → User (N : 1 관계)
+    users: Mapped["User"] = relationship(
```

Comment:
@Ryan-OH 
기덕님도 아래 남겨주신거 같은데 N : 1 관계 (user가 1) 이라면 여기 단수가 맞는거 같네요
```suggestion
    user: Mapped["User"] = relationship(
```

Reply:
현재 테이블명은 users로 되어있고, 앞부분은 속성명인데 그러면 user부분 속성명 전부 단수처리 할까요?

Reply:
@Ryan-OH 
네네!! 번거롭겠지만 그렇게 하는게 맞는거 같습니다 ㅠㅠ users 테이블 명에 상관없이 속성이름이므로 속성이 단수라면 단수로 하는게 맞는거 같습니닷..!